static/app/views/dashboardsV2/datasetConfig/base.tsx

```diff
@@ -11,5 +23,31 @@ export interface DatasetConfig<SeriesResponse, TableResponse> {
    * Transforms table API results into format that is used by
    * table and big number components
    */
-  transformTable?: (data: TableResponse) => TableData;
+  transformTable?: (
+    data: TableResponse,
```

Comment:
One thing I'm wondering about is if we would benefit more from sticking to positional args, or using an object so we can access args using keys. I think since we only have `contextualProps` as the optional arg right now this is probably fine and we can iterate if these start getting more complicated. Not really raising this as actionable feedback but to keep in mind as we build out more of these.